geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommand.java

```diff
@@ -0,0 +1,209 @@
+
+  static final String COMMAND_NAME = "alter query-service";
+  static final String COMMAND_HELP =
+      "Alter configuration parameters for the query service";
+  static final String METHOD_AUTHORIZER_NAME = "method-authorizer";
+  static final String METHOD_AUTHORIZER_NAME_HELP =
+      "The name of the class to be used for OQL method authorization";
+  static final String AUTHORIZER_PARAMETERS = "authorizer-parameters";
+  static final String AUTHORIZER_PARAMETERS_HELP =
+      "A comma separated list of all parameter values for the specified method authorizer. Requires '--"
+          + METHOD_AUTHORIZER_NAME + "' option to be used";
+  static final String SINGLE_AUTHORIZER_PARAMETER = "authorizer-parameter";
+  static final String SINGLE_AUTHORIZER_PARAMETER_HELP =
+      "A single parameter to be used by the specified method authorizer. Requires '--"
+          + METHOD_AUTHORIZER_NAME + "' option to be used";
+  static final String DEPRECATED_PROPERTY_ERROR = "Deprecated System Property: \"" + GEMFIRE_PREFIX
+      + "QueryService.allowUntrustedMethodInvocation\" is set to TRUE. In order to use a MethodInvocationAuthorizer, this property must be FALSE or undefined.";
+  static final String PARAMETERS_WITHOUT_AUTHORIZER_MESSAGE =
+      "The '--" + AUTHORIZER_PARAMETERS + "' and '--" + SINGLE_AUTHORIZER_PARAMETER
+          + "' options require '--"
+          + METHOD_AUTHORIZER_NAME + "' to be specified";
+  static final String SINGLE_PARAM_AND_PARAMETERS_SPECIFIED_MESSAGE =
+      "Only one of '--" + AUTHORIZER_PARAMETERS + "' and '--" + SINGLE_AUTHORIZER_PARAMETER
+          + "' options can be specified.";
+  static final String NO_ARGUMENTS_MESSAGE =
+      "No arguments were provided. No changes have been applied.";
+  static final String NO_MEMBERS_FOUND_MESSAGE = "No members found.";
+  static final String SECURITY_NOT_ENABLED_MESSAGE =
+      "Integrated security is not enabled for this distributed system. Updating the method authorizer requires integrated security to be enabled.";
+  public static final String PARTIAL_FAILURE_MESSAGE =
+      "In the event of a partial failure of this command, re-running the command or restarting failing members should restore consistency.";
+
+  @CliCommand(value = COMMAND_NAME, help = COMMAND_HELP)
+  @CliMetaData(
+      interceptor = "org.apache.geode.management.internal.cli.commands.AlterQueryServiceCommand$Interceptor")
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.MANAGE)
+  public ResultModel execute(
+      @CliOption(key = METHOD_AUTHORIZER_NAME,
+          help = METHOD_AUTHORIZER_NAME_HELP) String methodAuthorizerName,
+      @CliOption(key = AUTHORIZER_PARAMETERS,
+          help = AUTHORIZER_PARAMETERS_HELP) String[] authorizerParameters,
+      @CliOption(key = SINGLE_AUTHORIZER_PARAMETER,
+          help = SINGLE_AUTHORIZER_PARAMETER_HELP) String singleAuthorizerParameter) {
```

Comment:
We end up using a `Set<String>` internally, so I'd merge these two properties into just one as it might confuse users.

Reply:
It's necessary to provide an option that is parsed as a single String (as opposed to a String array) because if a user specifies `AUTHORIZER_PARAMETERS` with an argument that includes a comma as part of one of the parameters (for example, in a regex using the X{n,m} quantifier) then gfsh will split that argument on the comma as part of its parsing. That being said, if there's a way to ensure that user input will only be split on the "correct" commas then having only one option would definitely be preferable.

Reply:
Is this a correct statement - There is no way for a user to supply multiple regex's with commas in each regex.  This solution only allows for one regex with commas?

Reply:
@jhuynh1 That is correct. If there is a non-splitting comma in the argument, then the `authorizer-parameter` option must be specified, otherwise gfsh will treat that comma as a point to split the input string on.

Reply:
Is there any option to escape characters within a `gfsh` command parameter?.
I'd certainly prefer the above instead of having two different parameters for the same thing with just an `s` to differentiate them, it's counter intuitive and will probably cause issues for our users. If there's no way of achieving it, however, I suggest using two totally different and very verbose names for the parameters so users can tell the difference right away.

Reply:
We decided that the best solution to this issue is to split the gfsh input parameters on semicolons rather than commas using `optionContext = "splittingRegex=;"` in the `@CliOption` annotation for `AUTHORIZER_PARAMETERS`. This allows multiple regexes to be specified, even if they contain commas, and prevents the need for the corner-case `SINGLE_AUTHORIZER_PARAMETER` option.